src/prommis/nanofiltration/membrane_cascade_flowsheet/diafiltration_flowsheet_model.py

```diff
@@ -0,0 +1,1120 @@
+                                m.fs.stage[i],
+                                f'retentate_side_stream_{j}'
+                            )
+                        )
+                    )
+
+                    # fix recycle inlet of last stage mixers to LB (unused)
+                    if i == self.ns:
+                        m.fs.inlet_mixers[i, j].recycle.flow_vol[0].fix(1e-8)
+                        for sol in self.solutes:
+                            m.fs.inlet_mixers[i, j].recycle\
+                                                   .mass_solute[0, sol]\
+                                                   .fix(1e-8)
+
+        elif mixing == 'stage':
+            m.fs.inlet_mixers = Mixer(
+                pyo.RangeSet(self.ns),
+                inlet_list=['feed', 'diafiltrate', 'recycle'],
+                property_package=m.fs.properties,
+                material_balance_type=MaterialBalanceType.componentTotal,
+                energy_mixing_type=MixingType.none,
+                momentum_mixing_type=MomentumMixingType.none
+            )
+
+            # split the stage mixer into each stage side stream with splitter
+            m.fs.splitters = Splitter(
+                pyo.RangeSet(self.ns),
+                num_outlets=self.nt,
+                property_package=m.fs.properties,
+                material_balance_type=MaterialBalanceType.componentTotal,
+                momentum_balance_type=MomentumBalanceType.none,
+                energy_split_basis=EnergySplittingType.none
+            )
+            # m.fs.splitters = Splitter(
+            #     pyo.RangeSet(self.ns),
+            #     num_outlets=self.nt,
+            #     property_package=m.fs.properties,
+            #     material_balance_type=MaterialBalanceType.componentTotal,
+            #     momentum_balance_type=MomentumBalanceType.none,
+            #     energy_split_basis=EnergySplittingType.none
+            # )
+            # connect each split stream to a stage side stream
+            for i in pyo.RangeSet(self.ns):
+                m.fs.add_component(
+                    f'connect_inlet_splitter_{i}',
+                    Arc(
+                        source=m.fs.inlet_mixers[i].outlet,
+                        destination=m.fs.splitters[i].inlet
+                    )
+                )
+                # fix recycle inlet of last stage mixers to LB (unused)
+                if i == self.ns:
+                    m.fs.inlet_mixers[i].recycle.flow_vol[0].fix(1e-8)
+                    for sol in self.solutes:
+                        m.fs.inlet_mixers[i].recycle.mass_solute[0, sol]\
+                                                .fix(1e-8)
+                    # m.fs.inlet_mixers[i].recycle.mass_solute[0, 'Co']\
+                    #                             .fix(1e-8)
+                    # m.fs.inlet_mixers[i].recycle.mass_solute[0, 'Li']\
+                    #                             .fix(1e-8)
```

Comment:
Can these be removed?

Reply:
Yes, these were hard-coded for Co/Li and not needed after the addition of the for loop over all solutes.